src/component/addons/AddonList/AvailableAddons/AvailableAddons.tsx

```diff
@@ -21,40 +32,174 @@ interface IProvider {
 }
 
 interface IAvailableAddonsProps {
-    getAddonIcon: (name: string) => ReactElement;
     providers: IProvider[];
+    loading: boolean;
 }
 
 export const AvailableAddons = ({
     providers,
-    getAddonIcon,
+    loading,
 }: IAvailableAddonsProps) => {
     const navigate = useNavigate();
 
-    const renderProvider = (provider: IProvider) => (
-        <ListItem key={provider.name}>
-            <ListItemAvatar>{getAddonIcon(provider.name)}</ListItemAvatar>
-            <ListItemText
-                primary={provider.displayName}
-                secondary={provider.description}
-            />
-            <ListItemSecondaryAction>
-                <PermissionButton
-                    permission={CREATE_ADDON}
-                    onClick={() => navigate(`/addons/create/${provider.name}`)}
-                >
-                    Configure
-                </PermissionButton>
-            </ListItemSecondaryAction>
-        </ListItem>
+    const data = useMemo(() => {
+        if (loading) {
+            return Array(5).fill({
+                name: 'Provider name',
+                description: 'Provider description when loading',
+            });
+        }
+
+        return providers.map(({ name, displayName, description }) => ({
+            name,
+            displayName,
+            description,
+        }));
+    }, [providers, loading]);
+
+    const columns = useMemo(
+        () => [
+            {
+                id: 'Icon',
+                Cell: ({
+                    row: {
+                        original: { name },
+                    },
+                }: any) => (
+                    <Box
+                        data-loading
+                        sx={{
+                            pl: 2,
+                            pr: 1,
+                            display: 'flex',
+                            alignItems: 'center',
+                        }}
+                    >
+                        {getAddonIcon(name)}
+                    </Box>
+                ),
+            },
+            {
+                Header: 'Name',
+                accessor: 'name',
+                width: '90%',
+                Cell: ({
+                    row: {
+                        original: { name, description },
+                    },
+                }: any) => {
+                    return (
+                        <LinkCell
+                            data-loading
+                            title={name}
+                            subtitle={description}
+                        />
+                    );
+                },
+                sortType: 'alphanumeric',
+            },
+            {
+                Header: 'Actions',
+                id: 'Actions',
+                align: 'center',
+                Cell: ({ row: { original } }: any) => (
+                    <Box
```

Comment:
Would `ActionCell` work here? Could also extract a `CreateAddonButton` component for readability.